### PR TITLE
Adding optional prefix parameter to LOG functions

### PR DESCRIPTION
**Pull request by 12xx12**

### CODE REVIEW

File: Server/Plugins/APIDump/APIDesc.lua [hunk 1]
```diff
@@ -14696,6 +14696,29 @@ end
...
+				{
+					{
+						Params =
+						{
+							{
+								Name = "Message",
+								Type = "string",
+							},
+						},
+						Notes = "Logs a text into the server console and logfile using 'warning' severity (red text)",
```
Comment (bearbin) [inline Server/Plugins/APIDump/APIDesc.lua:14709]: Is this right? Looks like severity is just regular below not warning.

Comment (12xx12) [reply Server/Plugins/APIDump/APIDesc.lua:14709]: I thought I changed this :( I saw that too

Comment (12xx12) [reply Server/Plugins/APIDump/APIDesc.lua:14709]: Oh yeah: I changed it below but not there

**Comment by tigerw** [comment]
We can add an optional "Prefix" parameter to all the LOG functions, defaulted to the plugin name, allowing all the severity levels to be customised.

**Comment by 12xx12** [comment]
For now the sever checks if the second item on the stack is a string and adds this a prefix.

LOG("TEST") -> [MyPlugin] TEST
LOG("TEST", "") -> TEST
LOG("TEST", "TEST") -> TESTTEST

File: src/Bindings/ManualBindings.cpp [hunk 2]
```diff
@@ -353,16 +353,24 @@ Can take either a string or a cCompositeChat.
...
 	}
 
-	size_t len = 0;
-	const char * str = lua_tolstring(tolua_S, 1, &len);
-	Logger::LogSimple(fmt::format("[{}] {}", cManualBindings::GetLuaPlugin(tolua_S)->GetName(), std::string_view(str, len)), a_LogLevel);
+	auto Prefix = fmt::format("[{}] ", cManualBindings::GetLuaPlugin(tolua_S)->GetName());
+	if (lua_isstring(tolua_S, 2))
+	{
+		Prefix = lua_tolstring(tolua_S, 2, &len);
+	}
```
Comment (madmaxoft) [inline src/Bindings/ManualBindings.cpp:371]: Why doesn't this use `if`-`else`?

Comment (12xx12) [reply src/Bindings/ManualBindings.cpp:371]: One line of code less 🤷
Gonna change this 👍

Comment (12xx12) [reply src/Bindings/ManualBindings.cpp:371]: Nice to see you back again 😄

**Comment by 12xx12** [comment]
Presenting you: singed commits by me

**Comment by bearbin** [comment]
:fire: commits!

**Comment by tigerw** [comment]
smoking hot

Hold on a bit until Monday if you're waiting for my review, university is working us to the bone again.

**Comment by 12xx12** [comment]
Thanks I' gonna updated the plugins with a PR now

**Pull request merged**